datafusion/src/execution/context.rs

```diff
@@ -1220,6 +1233,74 @@ impl FunctionRegistry for ExecutionContextState {
     }
 }
 
+/// Task Context Properties
+pub enum TaskProperties {
+    ///SessionConfig
+    SessionConfig(SessionConfig),
+    /// KV pairs
+    KVPairs(HashMap<String, String>),
+}
+
+/// Task Execution Context
+pub struct TaskContext {
```

Comment:
`Task` is a vague term in the scope of DataFusion, I think. In the physical optimization phase, we would `repartition` plans based on `config.target_partition` when possible. Therefore, the task represents the initial partition of `DataSource`s? 

A plausible solution might require a major change on the current framework by introducing the `Stage` term in the DataFusion core. Then we could:
- Partition input data set based on conf like`max_bytes_per_partition`.
- processing data with physical operators serially for each input partition until we meet a "synchronization barrier" required by operators such as sort or aggregate.
- add an exchange operator (or repartition), and continue the computation in another task from the successor stage.

And by introducing `Stage`s into DataFusion core as well, we could make `task_id` required, make `task_context` the only parameter for `execute`.

For the current PR, I think we should articulate what does `Task` mean for DataFusion.

Reply:
+1. I'm a little confused, too.

Reply:
+1 for introducing `Stage` term in DataFusion. Ballista currently has `job_id`, `stage_id`, and `partition_id` (which perhaps could have been `task_id` instead).